### ISSUE TITLE
[ldap] new integration: service check and performance monitoring

### ISSUE THREAD

**Issue by ssbarnea** [DataDog/dd-agent#1250]
it would be very useful to be able to monitor LDAP servers for uptime and performance.
This could  be done by implementing a single type of test for all which by default would only try to connect, if credentials are specified would try a bind, and if a query is performed it would also perform the query and also register the total time needed.

**Comment by remh** [comment]
We have that in our backlog and will likely work on it in one of our next releases.

**Comment by olivielpeau** [comment]
Moving to 5.9.0 as the merge window for 5.8.0 is closing soon

**Comment by truthbk** [comment]
Unfortunately we didn't find the time to get this done for `5.9.0`, so pushing up to `5.10.0`, thank you for the patience.

**Comment by olivielpeau** [comment]
Moving back to triage until we can prioritize this for a future release. Feel free to comment here if this is a high priority feature for you.

**Comment by webratz** [comment]
would be glad to have that, especially a query and then counting the responses or storing the value of the response would be awesome

**Comment by mkporwit** [comment]
Yep, this would be handy to have. We're trying to unify our monitoring under Datadog, but we also use OpenLDAP for our auth system, so some sort of LDAP integration would be very handy.

**Comment by zacharystokes** [comment]
+1 - just had a request from our Infrastructure Services team this morning for this.

We use RedHat Directory Server, so even just a general overview of LDAP requests would be great.

**Comment by iancward** [comment]
Since this is a request for integration, this issue should probably be moved to https://github.com/DataDog/integrations-core/ or https://github.com/DataDog/integrations-extras. DataDog, do you know which one this might end up in?

**Comment by jeremy-lq** [comment]
@iancward We currently have this integration on our backlog, and have noted your interest. If released by Datadog it would go into the DataDog/integrations-core repo. If in the interim someone from the community writes one, it would go into the DataDog/integrations-extras repo (which is 100% maintained by the Datadog open source community).
